tests/e2e/watch_test.go

```diff
@@ -0,0 +1,470 @@
+			)
+			Expect(getMemoryLimit).To(ContainSubstring("700Mi"))
+			getMemoryRequest := runCmd("oc get dc openjdk-watch-" +
+				appTestName +
+				" -o go-template='{{range .spec.template.spec.containers}}{{.resources.requests.memory}}{{end}}'",
+			)
+			Expect(getMemoryRequest).To(ContainSubstring("400Mi"))
+		})
+
+		It("watch openjdk component created from local binary", func() {
+			runCmd("git clone " + openjdkURI + " " + tmpDir + "/binary/javalin-helloworld")
+			runCmd("mvn package -f " + tmpDir + "/binary/javalin-helloworld")
+			runCmd("odo create openjdk18 openjdk-watch-binary --binary " + tmpDir + "/binary/javalin-helloworld/target/javalin-hello-world-0.1-SNAPSHOT.jar --min-memory 400Mi --max-memory 700Mi")
+			runCmd("time odo push -v 4")
+			// Test multiple push so as to avoid regressions like: https://github.com/redhat-developer/odo/issues/1054
+			runCmd("time odo push -v 4")
+			runCmd("odo url create --port 8080")
+
+			startSimulationCh := make(chan bool)
+			go func() {
+				startMsg := <-startSimulationCh
+				if startMsg {
+					fmt.Println("Received signal, starting file modification simulation")
+
+					fileModificationPath := filepath.Join(
+						tmpDir,
+						"binary",
+						"javalin-helloworld",
+						"src",
+						"main",
+						"java",
+						"Application.java",
+					)
+
+					fileModificationCmd := fmt.Sprintf(
+						"sed -i 's/Hello World/Hello odo/g' %s",
+						fileModificationPath,
+					)
+					runCmd(fileModificationCmd)
+					runCmd("mvn package -f " + tmpDir + "/binary/javalin-helloworld")
+				}
+			}()
+			success, err := pollNonRetCmdStdOutForString(
+				"odo watch openjdk-watch-binary -v 4",
+				time.Duration(5)*time.Minute,
+				func(output string) bool {
+					url := runCmd("odo url list | grep `odo component get -q` | grep 8080 | awk '{print $2}' | tr -d '\n'")
+					curlOp := runCmd(fmt.Sprintf("curl %s", url))
+					return strings.Contains(curlOp, "Hello odo")
+				},
+				startSimulationCh,
+				func(output string) bool {
+					fmt.Println(output)
+					return strings.Contains(output, "Waiting for something to change")
+				},
+			)
+			Expect(success).To(Equal(true))
+			Expect(err).To(BeNil())
+
+			// Verify memory limits to be same as configured
+			getMemoryLimit := runCmd("oc get dc openjdk-watch-binary-" +
+				appTestName +
+				" -o go-template='{{range .spec.template.spec.containers}}{{.resources.limits.memory}}{{end}}'",
+			)
+			Expect(getMemoryLimit).To(ContainSubstring("700Mi"))
+			getMemoryRequest := runCmd("oc get dc openjdk-watch-binary-" +
+				appTestName +
+				" -o go-template='{{range .spec.template.spec.containers}}{{.resources.requests.memory}}{{end}}'",
+			)
+			Expect(getMemoryRequest).To(ContainSubstring("400Mi"))
+		})
+		It("watch wildfly component created from binary", func() {
+			runCmd("git clone " + wildflyURI + " " + tmpDir + "/binary/katacoda-odo-backend")
+			runCmd("mvn package -f " + tmpDir + "/binary/katacoda-odo-backend")
```

Comment:
there already is a war file in our tests suite examples (`tests/e2e/examples/binary/java/wildfly/ROOT.war`), would it make sense to use that, to speed up tests and skip building it every time?

Reply:
I think the point of the test here to be as real world as possible - so I propose we keep it.
On the other hand, if this application builds too slowly, we can definitely come up with a different one that will be faster

Reply:
I did it like this for 2 reasons:
1. As @geoand mentioned, the example I wanted it to be as close to real world example as possible
2. If I re-use the downloaded built binary, the next time I build it, I would have to copy it to that location right?

Reply:
ah, now I read the rest of the code, I see that you are using this source to make modification and verify that watch on binary works. It wouldn't make sense to use pre-build binary for this :-)